ramanujantools/matrix.py

```diff
@@ -74,19 +74,30 @@ def subs(self, substitutions: Dict) -> Matrix:
             return self.numerical_subs(substitutions)
         return self.xreplace(substitutions)
 
-    def _is_numeric(self, substitutions: Dict) -> bool:
+    def _can_call_flint_walk(self, trajectory: Dict, start: Dict) -> bool:
+        return (
+            self.free_symbols - set(start.keys()).union(trajectory.keys()) != set()
+            and trajectory.is_polynomial()
+            and start.is_polynomial()
+        )
+
+    def _can_call_numerical_walk(self, substitutions: Dict) -> bool:
```

Comment:
Also should receive both `trajectory` and `start` instead of this `substitutions`.

Reply:
And test this

Reply:
This is unused, I removed it